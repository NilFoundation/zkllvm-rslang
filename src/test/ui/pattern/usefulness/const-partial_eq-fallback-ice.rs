#![allow(warnings)]

struct MyType;

impl PartialEq<usize> for MyType {
    fn eq(&self, y: &usize) -> bool {
        true
    }
}

const CONSTANT: &&MyType = &&MyType;

fn main() {
    if let CONSTANT = &&MyType {
        //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
        println!("did match!");
    }
}
