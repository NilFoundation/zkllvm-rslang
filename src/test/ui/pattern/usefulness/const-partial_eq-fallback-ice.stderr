error: to use a constant of type `MyType` in a pattern, `MyType` must be annotated with `#[derive(PartialEq, Eq)]`
  --> $DIR/const-partial_eq-fallback-ice.rs:14:12
   |
LL |     if let CONSTANT = &&MyType {
   |            ^^^^^^^^

error: aborting due to previous error

